games-roguelike/FTL-gog/FTL-gog-1.6.9.25330.ebuild

```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit desktop unpacker
+
+DESCRIPTION="Top-down roguelike space ship simulator"
+HOMEPAGE="https://www.gog.com/game/faster_than_light"
+SRC_URI="ftl_advanced_edition_${PV//./_}.sh"
+RESTRICT="bindist fetch mirror strip test"
+
+LICENSE="GOG-EULA"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE=""
+
+DEPEND="app-arch/unzip"
+RDEPEND="media-libs/freetype
+	media-libs/libsdl[X,sound,opengl,video]
+	media-libs/libpng
+	sys-libs/zlib
+	virtual/opengl"
+
+S="${WORKDIR}/data/noarch"
+
+pkg_nofetch() {
+	elog "Please buy and download ${SRC_URI} from"
+	elog "https://www.gog.com/game/faster_than_light"
+}
+
+src_unpack() {
+	unpack_zip "${DISTDIR}/${A}"
```

Comment:
This is a legit question because I dont know and I want to learn: the distfile is .sh file? Whats the interaction here? :)

Reply:
The packaging that GOG uses for Linux builds is that they ship a shell script which kicks off a straightforward installer process when executed. After the shell script is the massive binary payload which is just a ZIP. `unzip` is perfectly capable of decompressing the file even though the first few hundred bytes are discarded; yes, it will echo a warning, but the first ebuild of this nature that I proposed had logic to work around this problem and it was suggested as extraneous. Ergo, emitting that warning is not offensive, it seems.

Reply:
Thanks! Thats a very clear explanation :)

With eapi-7, `app-arch/unzip` belongs to BDEPEND. After that, looks good to me! Its hard to test this package though.